pkg/apis/core/types.go

```diff
@@ -6493,3 +6508,19 @@ const (
 	// the destination set to the node's IP and port or the pod's IP and port.
 	LoadBalancerIPModeProxy LoadBalancerIPMode = "Proxy"
 )
+
+// OCIVolumeSource represents a OCI volume resource.
+type OCIVolumeSource struct {
```

Comment:
I think this should be ImageVolumeSource as well.

cc: @smarterclayton for thoughts.

Reply:
Renamed that as well together with the feature gate. It's a separate commit on top of this PR which can also be dropped again.

Reply:
I think `image` is the right name, given that we don't have a strong spec to reference for artifacts, while we do for behavior of images in Kubernetes.  

I added some suggested godoc edits to leave room for the container runtime implementation to add additional sources (as long as they are consistent with the behavior of container image mounting).  Single file mounting (vs directory) is the potentially only limitation I can see, but we could always define a CRI level transformation for any future OCI artifact spec representing single files or variation by adding new fields to the volume source to control a mapping from source -> directory.

I think we'd need a really strong reason to be inconsistent with container image, given we're expecting to reuse the CRI implementation of pull (same behavior), the fact that the vast majority of content here is going to be in the form of OCI images, and that users will have less to understand if image source pairs well with container image.

So +1 from me.

Reply:
Thanks, marking this point as resolved since we reached agreement.